extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,235 @@
+                {
+                    collectionBlob.Add(element);
+                }
+            }
+
+            var result = ToTargetCollectionType(targetType.GenericTypeArguments[0], collectionBlob);
+
+            return result;
+        }
+
+
+        private bool TryGetBindingDataContent(ModelBindingData bindingData, out IDictionary<string, string> bindingDataContent)
+        {
+            bindingDataContent = bindingData.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, string>(bindingData.Content.ToObjectFromJson<Dictionary<string, string>>(), StringComparer.OrdinalIgnoreCase),
+                _ => null
+            };
+
+            if (bindingDataContent is null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private async Task<object?> ToTargetType(Type targetType, string connectionString, string containerName, string blobName, ConverterContext context) => targetType switch
+        {
+            Type _ when targetType == typeof(String) => await GetBlobString(connectionString, containerName, blobName, context),
+            Type _ when targetType == typeof(Stream) => await GetBlobStream(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Byte[]) => await GetBlobBinaryData(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient) => CreateBlobReference<BlobClient>(connectionString, containerName, blobName, context),
+            Type _ when targetType == typeof(BlockBlobClient) => CreateBlobReference<BlockBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient) => CreateBlobReference<PageBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient) => CreateBlobReference<AppendBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobContainerClient) => CreateBlobContainerClient(connectionString, containerName),
+            _ => await DeserializeToTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private IEnumerable<object> ToTargetCollectionType(Type targetType, IEnumerable<object> array) => targetType switch
+        {
+            Type _ when targetType == typeof(BlobClient) => array.Select(p => (BlobClient)p),
+            Type _ when targetType == typeof(String) => array.Select(p => GetBlobContentString((BlobClient)p).GetAwaiter().GetResult()),
+            _ => throw new Exception()
+        };
+
+        private async Task<object?> DeserializeToTargetObject(Type targetType, string connectionString, string containerName, string blobName)
+        {
+            var content = await GetBlobString(connectionString, containerName, blobName);
+            return JsonConvert.DeserializeObject(content, targetType);
```

Comment:
I see we are using Newtonsoft.JSON.NET for deserialization here. Is that intentional? Wonder we should use the [serializer available on the worker option](https://github.com/Azure/azure-functions-dotnet-worker/blob/853fc0a0a8ff48464dda7b49aa7004410022fb12/src/DotNetWorker.Core/Hosting/WorkerOptions.cs#L18)?